### PR TITLE
Upgrade google-gax dependency

### PR DESCRIPTION
**Pull request by blowmage**
Upgrade all GRPC gems to use the new google-gax version. This change will require that all gems that have a dependency on GRPC to issue new releases.

The changes are:

* Remove direct dependencies on grpc and google-protobuf, use the dependencies declared in google-gax instead.
* Keep the direct dependency on googleapis-common-protos, since we want to ensure that minimum version for a bug fix.
* Upgrade Speech to use OperationsClient, this was renamed from OperationsApi in google-gax 0.7.0.

### CODE REVIEW

File: google-cloud-datastore/google-cloud-datastore.gemspec [hunk 1]
```diff
@@ -19,10 +19,8 @@ Gem::Specification.new do |gem|
   gem.required_ruby_version = ">= 2.0.0"
 
   gem.add_dependency "google-cloud-core", "~> 0.21.0"
-  gem.add_dependency "grpc", "~> 1.0"
-  gem.add_dependency "google-protobuf", "~> 3.0"
-  gem.add_dependency "googleapis-common-protos", "~> 1.3"
-  gem.add_dependency "google-gax", "~> 0.6.0"
+  gem.add_dependency "google-gax", "~> 0.7.1"
+  gem.add_dependency "googleapis-common-protos", "~> 1.3.5"
```
Comment (dazuma) [inline google-cloud-datastore/google-cloud-datastore.gemspec]: Should these be `"~> 1.3", ">= 1.3.5"`? As is, we're restricting the dependency more than it was before. (Not sure what our policy currently is with our dependencies.)

Comment (blowmage) [inline google-cloud-datastore/google-cloud-datastore.gemspec]: The existing dependency on `googleapis-common-protos` defined by `google-gax` is `~> 1.3.5`. So versions 1.4.0 or 1.5.0 will not resolve.

By specifying `~> 1.3.5` we are just slightly narrowing the acceptable version to ensure that it includes the file permission fix. If google-gax did this for us we would drop the dependency.

Comment (blowmage) [inline google-cloud-datastore/google-cloud-datastore.gemspec]: ^ Updated my comment to fix some typos.

Comment (dazuma) [inline google-cloud-datastore/google-cloud-datastore.gemspec]: Ah okay I forgot about the dependency that google-gax declares. SGTM. 👍

**Comment by swcloud** [comment]
I thought we're going to upgrade the dependency to google-gax as 0.8.0?

https://github.com/googleapis/gax-ruby/pull/57

**Comment by blowmage** [comment]
@swcloud We absolutely can upgrade to an upcoming 0.8.0 release. We haven't heard if 0.8.0 was coming, or when it would be released if it did, so we decided to get a head start on this now. By switching to 0.7.1 we can run the acceptance tests over the weekend. Making this change now doesn't mean we will be releasing new versions depending on 0.7.1.

**Pull request merged**